src/templates/TemplateProviderBase.ts

```diff
@@ -156,32 +157,37 @@ export abstract class TemplateProviderBase implements Disposable {
             throw new NotImplementedError('refreshProjKey', this);
         }
 
-        if (!this.sessionProjKey) {
-            this.sessionProjKey = await this.refreshProjKey();
+        if (!this._sessionProjKey) {
+            this._sessionProjKey = await this.refreshProjKey();
         }
 
-        return this.sessionProjKey;
+        return this._sessionProjKey;
     }
 
     public projKeyMayHaveChanged(): void {
         this._projKeyMayHaveChanged = true;
     }
 
-    public async hasProjKeyChanged(projKey: string | undefined): Promise<boolean> {
+    /**
+     * Returns true if the key changed
+     */
+    public async updateProjKeyIfChanged(projKey: string | undefined): Promise<boolean> {
+        let hasChanged: boolean;
         if (!this.refreshProjKey) {
```

Comment:
Do you need this conditional still?  If there is no refreshProjKey, I'd imagine the other conditions would all just be false and you'd hit the else statement.

Reply:
I like being explicit. Plus TypeScript would complain if I didn't have it
<img width="417" alt="Screen Shot 2021-03-03 at 7 04 17 PM" src="https://user-images.githubusercontent.com/11282622/109905216-4f548100-7c53-11eb-9478-b7b4bb701496.png">